File: WorkerServiceVB/Worker.vb
Imports Microsoft.Extensions.Hosting
Imports System.Threading

Public Class Worker
    Inherits BackgroundService
    Private ReadOnly _service As ActionService

    Public Sub New(service As ActionService)
        _service = service
    End Sub

    Protected Overrides Async Function ExecuteAsync(stoppingToken As CancellationToken) As Task
        While Not stoppingToken.IsCancellationRequested
            Await Task.Run(Sub() _service.Run("Worker"), stoppingToken)
        End While
    End Function
End Class

File: WorkerService/Worker.cs
namespace WorkerService;
public class Worker : BackgroundService
{
    private readonly ActionService _service;

    public Worker(ActionService service)
    {
        _service = service;
    }

    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
    {
        while (!stoppingToken.IsCancellationRequested)
        {
            await Task.Run(() => _service.Run("Worker"), stoppingToken);
        }
    }
}